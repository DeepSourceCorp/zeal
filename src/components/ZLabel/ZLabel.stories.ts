import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'
import ZLabel from './ZLabel.vue'

export default {
  title: 'Label',
  component: ZLabel,
  excludeStories: /.*Data$/
}

export const BasicLabels = () => ({
  components: { ZLabel },
  template: `<div class="padded-container flex gap-3 items-end">
      <z-label state="success">Analysis Active</z-label>
      <z-label state="error">Analysis Errored</z-label>
      <z-label state="warning">Analysis Incomplete</z-label>
      <z-label >Analysis Inactive</z-label>
    </div>`
})

export const LabelSuccess = () => ({
  components: { ZLabel },
  template: `<div class="padded-container flex gap-3 items-end">
      <z-label size="small" state="success" icon="check-circle">Analysis Active</z-label>
      <z-label size="base" state="success" icon="check-circle">Analysis Active</z-label>
      <z-label size="large" state="success" icon="check-circle">Analysis Active</z-label>
    </div>`
})

export const LabelError = () => ({
  components: { ZLabel },
  template: `<div class="padded-container flex gap-3 items-end">
      <z-label size="small" state="error" icon="x-circle">Analysis Errored</z-label>
      <z-label size="base" state="error" icon="x-circle">Analysis Errored</z-label>
      <z-label size="large" state="error" icon="x-circle">Analysis Errored</z-label>
    </div>`
})

export const LabelWarning = () => ({
  components: { ZLabel },
  template: `<div class="padded-container flex gap-3 items-end">
      <z-label size="small" state="warning" icon="minus-circle">Analysis Incomplete</z-label>
      <z-label size="base" state="warning" icon="minus-circle">Analysis Incomplete</z-label>
      <z-label size="large" state="warning" icon="minus-circle">Analysis Incomplete</z-label>
    </div>`
})

export const LabelDefault = () => ({
  components: { ZLabel },
  template: `<div class="padded-container flex gap-3 items-end">
      <z-label size="small" icon="circle">Analysis Inactive</z-label>
      <z-label size="base" icon="circle">Analysis Inactive</z-label>
      <z-label size="large" icon="circle">Analysis Inactive</z-label>
    </div>`
})
