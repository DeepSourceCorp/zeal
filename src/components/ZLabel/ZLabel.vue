<template>
  <span
    :class="`text-${baseColor} border-${baseColor} bg-${baseColor} ${sizeClass}`"
    class="inline-flex items-center gap-2 py-1 border bg-opacity-25"
  >
    <z-icon v-if="icon" :icon="icon" :size="iconSize" :color="`text-${baseColor}`"></z-icon>
    <slot></slot>
  </span>
</template>

<script>
import ZIcon from '../ZIcon/ZIcon.vue'
import Vue from 'Vue'

export default Vue.extend({
  name: 'ZLabel',
  components: {
    ZIcon
  },
  props: {
    icon: {
      type: String
    },
    state: {
      type: String,
      default: 'default',
      validator: function (value) {
        return ['success', 'error', 'warning', 'default'].includes(value)
      }
    },
    size: {
      type: String,
      default: 'base',
      validator: function (value) {
        return ['small', 'base', 'large'].includes(value)
      }
    }
  },
  computed: {
    baseColor() {
      if (this.state === 'success') return 'juniper'
      else if (this.state == 'error') return 'cherry'
      else if (this.state == 'warning') return 'honey'
      else return 'vanilla-100'
    },
    sizeClass() {
      if (this.size === 'small') return 'rounded-sm text-xs px-2'
      else if (this.size === 'large') return 'rounded-sm text-md px-3 py-2'
      return 'rounded-sm text-sm px-2'
    },
    iconSize() {
      if (this.size === 'large') return ''
      else return 'small'
    }
  }
})
</script>
