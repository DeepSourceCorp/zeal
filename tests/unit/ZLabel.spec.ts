import ZLabel from '../../src/components/ZLabel'

import { shallowMount } from '@vue/test-utils'

const slotData = {
  default: 'Hello World'
}

describe('Labels', () => {
  it('renders without icon', () => {
    const wrapper = shallowMount(ZLabel, {
      slots: slotData
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with icon', () => {
    const wrapper = shallowMount(ZLabel, {
      propsData: {
        icon: 'check-circle'
      },
      slots: slotData
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with state', () => {
    expect(
      shallowMount(ZLabel, {
        propsData: {
          icon: 'check-circle',
          state: 'success'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZLabel, {
        propsData: {
          icon: 'check-circle',
          state: 'error'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZLabel, {
        propsData: {
          icon: 'check-circle',
          state: 'warning'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZLabel, {
        propsData: {
          icon: 'check-circle',
          state: 'default'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()
  })

  it('renders with size', () => {
    expect(
      shallowMount(ZLabel, {
        propsData: {
          icon: 'check-circle',
          size: 'small'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZLabel, {
        propsData: {
          icon: 'check-circle',
          size: 'base'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZLabel, {
        propsData: {
          icon: 'check-circle',
          size: 'large'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()
  })
})
