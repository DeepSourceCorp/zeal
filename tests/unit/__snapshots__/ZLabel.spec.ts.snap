// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labels renders with icon 1`] = `<span class="inline-flex items-center uppercase font-semibold border rounded-full bg-ink-400 border-slate-400 text-vanilla-200 tracking-wide leading-none text-xxs px-1 py-1 pr-2 space-x-1"><z-icon-stub icon="solid-circle" size="small" position="" color="robin"></z-icon-stub> <span>Hello World</span></span>`;

exports[`Labels renders with size 1`] = `<span class="inline-flex items-center uppercase font-semibold border rounded-full bg-ink-400 border-slate-400 text-vanilla-200 tracking-wide leading-none text-xxs px-1 py-1 pr-2 space-x-1"><z-icon-stub icon="solid-circle" size="small" position="" color="robin"></z-icon-stub> <span>Hello World</span></span>`;

exports[`Labels renders with size 2`] = `<span class="inline-flex items-center uppercase font-semibold border rounded-full bg-ink-400 border-slate-400 text-vanilla-200 tracking-wide leading-none text-xs px-1 py-1 pr-3 space-x-2"><z-icon-stub icon="solid-circle" size="medium" position="" color="robin"></z-icon-stub> <span>Hello World</span></span>`;

exports[`Labels renders with state 1`] = `<span class="inline-flex items-center uppercase font-semibold border rounded-full bg-ink-400 border-slate-400 text-vanilla-200 tracking-wide leading-none text-xxs px-1 py-1 pr-2 space-x-1"><z-icon-stub icon="check-circle" size="small" position="" color="juniper"></z-icon-stub> <span>Hello World</span></span>`;

exports[`Labels renders with state 2`] = `<span class="inline-flex items-center uppercase font-semibold border rounded-full bg-ink-400 border-slate-400 text-vanilla-200 tracking-wide leading-none text-xxs px-1 py-1 pr-2 space-x-1"><z-icon-stub icon="x-circle" size="small" position="" color="cherry"></z-icon-stub> <span>Hello World</span></span>`;

exports[`Labels renders with state 3`] = `<span class="inline-flex items-center uppercase font-semibold border rounded-full bg-ink-400 border-slate-400 text-vanilla-200 tracking-wide leading-none text-xxs px-1 py-1 pr-2 space-x-1"><z-icon-stub icon="alert-circle" size="small" position="" color="honey"></z-icon-stub> <span>Hello World</span></span>`;

exports[`Labels renders with state 4`] = `<span class="inline-flex items-center uppercase font-semibold border rounded-full bg-ink-400 border-slate-400 text-vanilla-200 tracking-wide leading-none text-xxs px-1 py-1 pr-2 space-x-1"><z-icon-stub icon="solid-circle" size="small" position="" color="robin"></z-icon-stub> <span>Hello World</span></span>`;

exports[`Labels renders without icon 1`] = `<span class="inline-flex items-center uppercase font-semibold border rounded-full bg-ink-400 border-slate-400 text-vanilla-200 tracking-wide leading-none text-xxs px-1 py-1 pr-2 space-x-1"><z-icon-stub icon="solid-circle" size="small" position="" color="robin"></z-icon-stub> <span>Hello World</span></span>`;
